File: A_IQ_test.py
def get_number(n: int) -> int:
    even_numbers_indexes: list[int] = []
    odd_numbers_indexes: list[int] = []

    numbers: list[int] = list(map(int, input().split()))

    for index, number in enumerate(numbers):
        if number % 2 == 0:
            even_numbers_indexes.append(index)
        else:
            odd_numbers_indexes.append(index)       

    if len(even_numbers_indexes) == 1:
        return even_numbers_indexes[0] + 1
    else:
        return odd_numbers_indexes[0] + 1

def main() -> None:
    n: int = int(input())
    print(get_number(n))    

if __name__ == "__main__":
    main()

File: A_IQ_test.rs
use std::io::BufRead;

fn read_line() -> String {
    let mut stdin = std::io::stdin().lock();
    let mut line = String::new();
    stdin.read_line(&mut line).unwrap();
    line
}

fn get_number(n: u8) -> u8 {
    let mut even_number_indexes: Vec<u8> = Vec::new();
    let mut odd_number_indexes: Vec<u8> = Vec::new();

    let numbers: Vec<u8> = read_line()
        .split_ascii_whitespace()
        .map(|s| s.parse().unwrap())
        .collect();

    for (index, number) in numbers.iter().enumerate() {
        if number % 2 == 0 {
            even_number_indexes.push(index as u8 + 1);
        } else {
            odd_number_indexes.push(index as u8 + 1);
        }
    }

    if even_number_indexes.len() == 1 {
        even_number_indexes[0]
    } else if odd_number_indexes.len() == 1 {
        odd_number_indexes[0]
    } else {
        0
    }
}

fn main() {
    let n: u8 = read_line().trim().parse().unwrap();
    let result = get_number(n);
    println!("{}", result);
}